### DISCUSSION TITLE
Make other VSTs be in sync with Playtime when REAPER transport is NOT RUNNING

### DISCUSSION THREAD

**Discussion by helgoboss** [helgoboss/helgobox#1022] [Ideas]
When REAPER transport is paused, Playtime uses its own timeline and playback position in order to not be influenced by play/edit cursor changes in the arrangement. Some VST instruments, e.g. drum computers, sync with REAPER's current position which is not the same as Playtime's position. Find a way to deal with this.

**Comment by helgoboss** [comment]
Request for feedback: I see many people voted for this, but why? Which VSTs do you actually have in mind that would benefit from this and how? What are your expectations here?

I had XLNAudio XO in mind initially but it looks like it wouldn't benefit from it because it's not even trying to sync to the DAW when the DAW is not playing.

**Comment by Chianti** [comment]
> Which VSTs do you actually have in mind that would benefit from this

I wasn't part of any original discussion about this, but I'm guessing maybe the Toontrack line?
EZ Drummer, EZ Bass and EZ Keys all have their own time-lined midi arrangers built in, which sync to the Reaper timeline.

At first thought, a non-technical person like me could fuzzily conceive of trying to do 'hybrid' sketches/arrangements, using Toontrack's editors for all the midi stuff, and recording guitar and vocals audio into Playtime2 slots, trying to build up an arrangement using scenes, so that the scenes follow through the Reaper timeline, in sync with the Toontrack arrangers.

If you pause Reaper the Toontrack plugins and Playtime2 clips also pause, because they're all synced, But when you play again, whilst Reaper and Toontrack continue from where they were, Playtime2 starts the clips again from the beginning, putting it out of sync.

The 'why' part of this is because the Toontrack midi arrangers have distinct advantages. For instance, you can right click a midi part in the arranger, and look for similar parts in your library, or which folder/family the part is from. You can very easily tell the plugin to intelligently make the part more or less complex, or louder and softer, or push/pull the rhythm, without any step editing at all. But once the midi part is outside of the Toontrack environment, you lose all of that, so that's why people might be reluctant to drag the Toontrack midi into the Playtime slots.

But never mind plugins - wouldn't this same issue occur if you had some audio/midi in the Reaper timeline, and some of your arrangement in Playtime2? If you paused Reaper and then restarted, wouldn't Playtime become out of sync for the same reason - starting the clips from the beginning instead of where they were paused?

I don't know what the answer to this is, nor what the desired behaviour should be. Maybe it just goes against the whole concept of a clip arranger, and doesn't have a solution?

------------
In addition to this 'pause' and restart out of sync' issue described, I'm having difficulty with the concept of 'manually playing through a sequence of scenes whilst Reaper is playing though its timeline'. In other words, I'm trying to set-off Reaper and 'Playtime Scene1' synced together, and while that scene is playing, clicking another scene (let's say Scene 3) to cue it to start playing automatically _after_ Scene 1 finishes. Then while Scene 3 is playing, clicking Scene 8 to start playing after Scene 3 finishes. I thought this would be easy and intuitive, but it doesn't seem possible?

One thing preventing this workflow with Reaper might be that 'full transport synchronisation' is not be working properly? I can't seem to get Playtime and Reaper to start playing together with the click of one button?

But even without trying to sync to Reaper, I don't seem to be able to manually skip from scene to scene sequentially. It might be that exclusive mode is not working properly? For example, when I click Scene 3, it plays _at the same time_ as Scene 1, even though the columns are set to be exclusive (so Cell 3 should not be able to play at the same time as Cell 1 in any of the columns?). There also doesn't appear to be a start trigger which stipulates 'play after current scene has finished'.

In the same vein, I'm also having trouble using the Matrix Sequencer due to this - it's like I'm supposed to set the 'start timing' to 1 bar or immediate, and click the new scene button _exactly_ at the right time, rather than having it cued-up to play automatically when the current scene ends? I'm a little confused, because I thought clicking around to get clips and scenes to play one after the other was one of the strengths of using a clip arranger?

I definitely need to watch your tutorial video again, and maybe some of the latter answers will become apparent. I also still need to play about with it a lot more to try to solidify in my mind the intended usage and workflow, rather than trying to make Playtime do things it wasn't designed to do.

**Reply by helgoboss** [reply]
> If you pause Reaper the Toontrack plugins and Playtime2 clips also pause, because they're all synced, But when you play again, whilst Reaper and Toontrack continue from where they were, Playtime2 starts the clips again from the beginning, putting it out of sync.

Hmm. This is something else, not related to what I meant by the title ("Make other VSTs be in sync with Playtime when REAPER transport stopped"). REAPER transport is playing in your case. In this case, Playtime will do its start/stop quantization in relation to REAPER's project timeline. This is already working now! Yes, the clips always start from the beginning, but this is how it's supposed to be. Is it possible that you have very long clips, not looped? I mean, if it's short looped stuff, it shouldn't matter. If it does, then you probably want a new clip trigger mode: #1038 

> But never mind plugins - wouldn't this same issue occur if you had some audio/midi in the Reaper timeline, and some of your arrangement in Playtime2? If you paused Reaper and then restarted, wouldn't Playtime become out of sync for the same reason - starting the clips from the beginning instead of where they were paused?

Ah wait, you are talking about "Pause" button? Hmm. Please create a new feature request for that (something like "Letting REAPER pause pause Playtime clips"), if you are interested in that.

> In addition to this 'pause' and restart out of sync' issue described, I'm having difficulty with the concept of 'manually playing through a sequence of scenes whilst Reaper is playing though its timeline'. In other words, I'm trying to set-off Reaper and 'Playtime Scene1' synced together, and while that scene is playing, clicking another scene (let's say Scene 3) to cue it to start playing automatically _after_ Scene 1 finishes. Then while Scene 3 is playing, clicking Scene 8 to start playing after Scene 3 finishes. I thought this would be easy and intuitive, but it doesn't seem possible?

That's a matter of clip start timing. A scene itself doesn't have any length. When you trigger a scene, all clips in that scene are triggered for playback according to their start timing, e.g. they might obey to the global start timing of 4 bars. Then, when you press another scene, it will wait 4 bars and then trigger the others clips in that other scene.

> One thing preventing this workflow with Reaper might be that 'full transport synchronisation' is not be working properly? I can't seem to get Playtime and Reaper to start playing together with the click of one button?

Which button? You might have run into the bug #1000, that will be fixed in 2.16.1.
 
> But even without trying to sync to Reaper, I don't seem to be able to manually skip from scene to scene sequentially. It might be that exclusive mode is not working properly? For example, when I click Scene 3, it plays _at the same time_ as Scene 1, even though the columns are set to be exclusive (so Cell 3 should not be able to play at the same time as Cell 1 in any of the columns?).

Did this happen only a few times or always? If only a few times, you might have run into bug #967, that will be fixed in 2.16.1 as well. If it always happens, then maybe you have some clips in there that don't override the global start or stop timing? This is what counts!

> There also doesn't appear to be a start trigger which stipulates 'play after current scene has finished'.

Because scenes don't have a concept of length. Scenes are just triggers of a row of clips.

> In the same vein, I'm also having trouble using the Matrix Sequencer due to this - it's like I'm supposed to set the 'start timing' to 1 bar or immediate, and click the new scene button _exactly_ at the right time, rather than having it cued-up to play automatically when the current scene ends? I'm a little confused, because I thought clicking around to get clips and scenes to play one after the other was one of the strengths of using a clip arranger?

Hmm. Same thing here. Everything depends on the clip start/stop timing settings.

**In the future, please use a separate discussion for usage questions, in the Q&A category. Thanks!**

**Comment by helgoboss** [comment]
I just renamed this idea.

Old title: "Make other VSTs be in sync with Playtime when REAPER transport stopped"

New title: "Make other VSTs be in sync with Playtime when REAPER transport NOT RUNNING"

Hopefully makes it more clear. It's really just about the situation when REAPER is not playing. In that case, REAPER has an imaginary play cursor that continuously moves forward until e.g. changing tempo. VSTs with rhythmical alignment will listen to that play cursor. But Playtime uses its own play cursor (in order to not be disrupted by tempo changes).

I bet this FR wouldn't have that many votes if people knew what it really means ;) It's probably only edge cases where this matters.

**Comment by mortalsound** [comment]
Good day! 
I think the ability to synchronize cursor position in rhythmically dependent plugins is very necessary. 
For example, apart from the above described drum machines and all sorts of EZ-products, I use multi-processing for lives like Effectrix and looperator. And they don't work without running them in the arrangement playback window.

**Comment by Ian1971** [comment]
I would love this. The only work around right now is to press play in reaper and its not ideal.
My use case is a kontakt drum bank I have that has an audition button, this is out of sync if I press it, unless reaper is playing (but then midi editing in playtime is weird). The other use case is reaktor drum machines which need to be in sync, again the only option currently is to press play in reaper.

**Comment by punkUser** [comment]
Possibly related but not sure... the one case I've run into where Reaper transport "playing" seems to matter is when using BlueARP (https://omg-instruments.com/wp/?page_id=63). It will tend to work ok initially even when Reaper isn't playing, but if you leave something playing for a while it will start to desynchronize with the rest of playback. None of the "reset" options in BlueARP prevent this from happening, but having Reaper playback running *does* prevent it. 

Thus I tend to use "full synchronization" but there are various downsides to that as well:
1) it's a bit messy when you have items on the actual timeline (ex. after exporting a matrix sequence)
2) it doesn't start reaper playback if you click a clip launch button as the first thing you do... you have to explicitly click the playtime "play" button and then click a clip (which can be a bit annoying as it will launch whatever previous clips were playing and/or start a new synchronization window which you have to wait until the end of to launch your first clip)
3) there's no visibility into whether reaper is playing if Playtime is full-screen, so due to the above you can still get into states where BlueARP will drift over time and not necessarily notice
4) it gets really awkward if there are any automations on the associated reaper tracks... normally I'd just click past the end of all the reaper stuff but sometimes the automations will then put parameters to unusable values there. (This one seems generally awkward and not directly related to Playtime/reaper transport sync but also breaks a bit of the feeling that the playtime cursor is separate from reaper playback.)
4) tempo change scenes are presumably unreliable

If this is unrelated I can move it to another issue, but this is the closest discussion I've found to the issue I've run into here.

**Comment by declanhalpin** [comment]
UPVOTING!

I use VCV RACK and I'm looking integrate Playtime with it.  VCV needs a proper Reaper "Start" to activate clocks, sequencers, arps, etc.

**Comment by infeasibler** [comment]
I have a clunky workaround to this. A reascript that spots if anything is playing and starts the transport. It's far from perfect but it works for me. [here](https://github.com/infeasibler/reaperplugins/blob/main/playtime_bridge/playtime_bridge.lua)

Drop it in reaper scripts folder. Then you can add to to toolbar as normal. Toggle it on when you want it active.
It's not totally smooth when it detects and starts but its better for me than not having it.
When audio is silent it stops the transport

In fact it doesn't need playtime at all if for some reason you wanted reaper to auto start transport when there was audio playing it can fit that use case too.

**Reply by Chianti** [reply]
Thank you for sharing. Much appreciated!

I'll try your script if I find time or reason to go back to Playtime and mess around with it again, but I kind of moved on from it. I'm glad I was able to support HB projects with the purchase of Playtime 1/2, but I'm not quite clever enough to understand how to use Playtime to assist with arrangement and auditioning of Rock/Stoner/Metal parts. I had imagined it was like a sketchpad that could be used before committing to a proper performance in the arrange window, but I was unable to record parts into it successfully, nor work out how to quickly sequence auditioning combinations of parts.

I started using BirdBird's Global Sampler to build little sketches in the Arrange Window, using one track set to free item positioning. Not really happy with that approach either. So the search isn't over! :)

**Reply by infeasibler** [reply]
Playtime can do those things you describe
 I use it for jamming. Record a riff I like. Add a bass whilst it's looping. Add drums. Synth. Then develop more scenes. Eventually recording an arrangement which exports to a standard reaper tracks. From there turn it into a full song. Everything works out of the box for me except for drum auditioning the drum vsts need to be in sync which means reaper transport needs to be playing.
